manifests/init.pp

```diff
@@ -0,0 +1,57 @@
+# @summary Profile for installing and managing Java.
+#
+# Profile for installing and managing Java.
+#
+# @param install_lock_files
+#   Boolean - Install lock files or not.
+#
+# @param lock_files
+#   Hash of lock files (and parent dir(s)).
+#
+# @param pkg_list
+#   Hash of package names to install
+#   Keys must be an OS Family name
+#   Values must be a Hash of the format:
+#       Key = package name
+#           A package name can be prefixed with "-" to remove it from the list.
+#           In this way, a higher level hiera layer can remove a package added
+#           by a lower, common layer.
+#
+#       Value (optional):
+#           1. <EMPTY> in which case, the package will be installed
+#           2. <HASH> conataining valid attributes for the 
+#              "package" native puppet resource type
+#
+# @example
+#   include profile_java
+class profile_java (
+  Boolean $install_lock_files,
+  Hash    $lock_files,
+  Hash    $pkg_list,
+) {
+  # Default value (for bare package names)
+  $default = { 'ensure' => 'installed' }
+
+  # Limit list of packages by OS Family
+  $packages = $pkg_list[$facts['os']['family']]
+
+  if $packages =~ Hash[String[1], Data, 1] {
+    # Find keys without a value and add default value
+    $sane_pkg_list = $packages.map |$key, $val| {
+      if $val { [$key, $val] }
+      else { [$key, $default] }
+    }.convert_to(Hash)
+
+    ensure_packages( $sane_pkg_list )
+  }
+
+  if $install_lock_files {
+    File {
+      group => 'root',
+      owner => 'root',
+    }
+    $lock_files.each | $k, $v | {
+      file { $k: * => $v }
+    }
```

Comment:
This approach works, but I would generally do this instead with an `ensure_resources` of the `lock_files` hash.

The main advantage is it deals better with potential duplicates, like the `/etc` directory.

Reply:
Done!